### PR TITLE
Update a few mutation types 

### LINKED ISSUE #1211: Fix the mutation input types
**Describe the bug**

https://github.com/PalisadoesFoundation/talawa-api/pull/1168#issuecomment-1465270002

Accept a single argument named input of type UpdateOrganizationInput, include the organization's id within that and make the id required. Also make the input required in mutation definition like this input: UpdateOrganizationInput!. It doesn't make sense to make a mutation request without the input object.

Mutation receiving a single argument named input is the convention in graphql. It's a good practice to stick to conventions.

Fixes the types and refactors the arguments for `updateOrganization` , `createPost` and other mutations when needed as directed by @xoldyckk  
**To Reproduce**

1.
2.
3.
4.

**Expected behavior**

**Actual behavior**
A clear and concise description of how the code performed w.r.t expectations.

**Screenshots**

**Additional details**

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by kb-0311**
**What kind of change does this PR introduce?**

**Issue Number:**

Fixes #1211 

**Did you add tests for your changes?**

**Snapshots/Videos:**

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

### CODE REVIEW

**Comment by kb-0311** [comment]
This will also have a sister issue associated with it. Once the changes are finalised

@xoldyckk I noticed that we can make these changes to a lot of other mutation types.

What all other changes can be made to mutation types?

**Comment by anshgoyalevil** [comment]
> ## Manage this branch in [Squash](https://app.squash.io/accounts/)
> Test this branch here: https://kb-0311updateorgtype-j86fy.squash.io

Ignore this comment for now. It is a new feature which would be live shortly.

**Comment by xoldyckk** [comment]
@kb-0311 There are some resolvers which are not defined in schema, idk what they are for.

**Comment by kb-0311** [comment]
@xoldyckk Pls do check out the structure of arguments in the mutation resolvers.

If it is okay I will make the following changes in the unit tests and client apps.

**Comment by xoldyckk** [comment]
@kb-0311 input name for many of the mutation fields is still not correct. Please re check carefully.

**Comment by kb-0311** [comment]
@xoldyckk Yeah changed them , pls do check it out.

File: src/typeDefs/mutations.ts [hunk 1]
```diff
@@ -5,187 +5,183 @@ import { gql } from "apollo-server-core";
...
-    adminRemoveGroup(groupId: ID!): Message! @auth
+    adminRemoveGroup(input: AdminRemoveGroupInput!): Message! @auth
 
     assignUserTag(input: ToggleUserTagAssignInput!): User @auth
 
-    blockPluginCreationBySuperadmin(userId: ID!, blockUser: Boolean!): User!
-      @auth
-      @role(requires: SUPERADMIN)
+    blockPluginCreationBySuperadmin(
+      input: blockPluginCreationBySuperadminInput!
```
Comment (xoldyckk) [inline src/typeDefs/mutations.ts:31]: Capitalize the input name

File: src/typeDefs/mutations.ts [hunk 2]
```diff
@@ -5,187 +5,183 @@ import { gql } from "apollo-server-core";
...
-    signUp(data: UserInput!, file: String): AuthData!
+    signUp(input: SignUpInput!): AuthData!
 
-    togglePostPin(id: ID!): Post! @auth
+    togglePostPin(input: TogglePostPinInput!): Post! @auth
 
     unassignUserTag(input: ToggleUserTagAssignInput!): User @auth
 
-    unblockUser(organizationId: ID!, userId: ID!): User! @auth
+    unblockUser(input: UserAndOrganizationInput!): User! @auth
```
Comment (xoldyckk) [inline src/typeDefs/mutations.ts:155]: You forgot to change the input name here.

File: src/typeDefs/inputs.ts [hunk 3]
```diff
@@ -2,27 +2,152 @@ import { gql } from "apollo-server-core";
...
   }
 
+  input CreateMessageChatInput {
+    message: String!
+    receiver: ID!
+  }
+
+  input CreateOrganizationInput {
+    data: OrganizationInput!
+    file: String
```
Comment (xoldyckk) [inline src/typeDefs/inputs.ts:120]: Change these names like `file` to what they actually represents, for example `image`, `video`, `profile_picture` etc., whatever's accurate.

**Comment by kb-0311** [comment]
@xoldyckk Added the changes, the unblock user one went past my eye.

Even if you approach pls do not merge this one as first another sister issue will need to be created on the client side to implement these changes in the schema.

**Comment by xoldyckk** [comment]
@kb-0311 Naming input fields the same as they're written in database would make it easier and cleaner to substitute the values in resolvers.

**Comment by kb-0311** [comment]
@palisadoes I will be busy for next few weeks till 10th may because of academic practicals , end sem exams and academic projects but will pick up the pace after 10th may. Sorry for the delay.

**Comment by kb-0311** [comment]
Closing this for now because of too many conflicting files can be  a good first issue for other contributors.

**Pull request closed without merging**